File: src/main/java/com/hirehub/SpringServer/Entity/Candidate.java
package com.hirehub.SpringServer.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
@Entity
@Table(name = "candidates")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Candidate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long candidateId;

    @OneToOne
    @JoinColumn(name = "userId")
    private User user;

    private String degree;
    private Integer experienceYears;

    @Column(columnDefinition = "TEXT")
    private String skills;

    private String resumeUrl;

    @Column(columnDefinition = "TEXT")
    private String education;

    @Column(columnDefinition = "TEXT")
    private String experience;

    @Column(columnDefinition = "TEXT")
    private String certifications;

    @Column(columnDefinition = "TEXT")
    private String projects;

    private String currentJobTitle;
    private String currentCompany;
    private Integer currentSalary;
    private Integer expectedSalary;
    private String noticePeriod;

    private String resumeHeadline;

    @Column(columnDefinition = "TEXT")
    private String summary;

    private String linkedinUrl;
    private String githubUrl;
    private String portfolioUrl;

    @Column(columnDefinition = "TEXT")
    private String preferredLocations;

    private String employmentType;
    private String workType;

    private Timestamp createdAt;
    private Timestamp updatedAt;

    public Candidate(Candidate c){
        this.candidateId = c.candidateId;
        this.user = c.user;
        this.degree = c.degree;
        this.experienceYears = c.experienceYears;
        this.skills = c.skills;
        this.resumeUrl = c.resumeUrl;
        this.education = c.education;
        this.experience = c.experience;
        this.certifications = c.certifications;
        this.projects = c.projects;
        this.currentJobTitle = c.currentJobTitle;
        this.currentCompany = c.currentCompany;
        this.currentSalary = c.currentSalary;
        this.expectedSalary = c.expectedSalary;
        this.noticePeriod = c.noticePeriod;
        this.resumeHeadline = c.resumeHeadline;
        this.summary = c.summary;
        this.linkedinUrl = c.linkedinUrl;
        this.githubUrl = c.githubUrl;
        this.portfolioUrl = c.portfolioUrl;
        this.preferredLocations = c.preferredLocations;
        this.employmentType = c.employmentType;
        this.workType = c.workType;
        this.createdAt = c.createdAt;
        this.updatedAt = c.updatedAt;
    }
}

File: DotNetServer/Models/Entities/Candidate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace HireHubServer.Models.Entities;

[Table("candidates")]
public class Candidate
{
  [Key]
  [Column("candidate_id")]
  public long CandidateId { get; set; }

  // 🔑 FK → users.user_id
  [Column("user_id")]
  public long UserId { get; set; }

  // Navigation
  public User User { get; set; } = null!;

  [Column("degree")]
  public string? Degree { get; set; }

  [Column("experience_years")]
  public int? ExperienceYears { get; set; }

  [Column("skills")]
  public string? Skills { get; set; }

  [Column("resume_url")]
  public string? ResumeUrl { get; set; }

  [Column("education")]
  public string? Education { get; set; }

  [Column("experience")]
  public string? Experience { get; set; }

  [Column("certifications")]
  public string? Certifications { get; set; }

  [Column("projects")]
  public string? Projects { get; set; }

  [Column("current_job_title")]
  public string? CurrentJobTitle { get; set; }

  [Column("current_company")]
  public string? CurrentCompany { get; set; }

  [Column("current_salary")]
  public int? CurrentSalary { get; set; }

  [Column("expected_salary")]
  public int? ExpectedSalary { get; set; }

  [Column("notice_period")]
  public string? NoticePeriod { get; set; }

  [Column("resume_headline")]
  public string? ResumeHeadline { get; set; }

  [Column("summary")]
  public string? Summary { get; set; }

  [Column("linkedin_url")]
  public string? LinkedinUrl { get; set; }

  [Column("github_url")]
  public string? GithubUrl { get; set; }

  [Column("portfolio_url")]
  public string? PortfolioUrl { get; set; }

  [Column("preferred_locations")]
  public string? PreferredLocations { get; set; }

  [Column("employment_type")]
  public string? EmploymentType { get; set; }

  [Column("work_type")]
  public string? WorkType { get; set; }

  [Column("created_at")]
  public DateTime CreatedAt { get; set; }

  [Column("updated_at")]
  public DateTime? UpdatedAt { get; set; }

  // Navigation
  [JsonIgnore]
  public ICollection<Referral> Referrals { get; set; } = new List<Referral>();
}
